lib/std/math.zig

```diff
@@ -621,6 +621,59 @@ fn testDivFloor() void {
     testing.expect((divFloor(f32, -5.0, 3.0) catch unreachable) == -2.0);
 }
 
+pub fn divCeil(comptime T: type, numerator: T, denominator: T) !T {
+    @setRuntimeSafety(false);
+    if (comptime std.meta.trait.isNumber(T) and denominator == 0) return error.DivisionByZero;
+    const info = @typeInfo(T);
+    switch (info) {
+        .ComptimeFloat, .Float => return @ceil(numerator / denominator),
+        .ComptimeInt, .Int => {
+            if (numerator < 0 and denominator < 0) {
+                if (info == .Int and numerator == minInt(T) and denominator == -1)
+                    return error.Overflow;
+                return @divFloor(numerator + 1, denominator) + 1;
+            }
+            if (numerator > 0 and denominator > 0)
+                return @divFloor(numerator - 1, denominator) + 1;
+            return @divTrunc(numerator, denominator);
+        },
+        else => @compileError("divCeil unsupported on " ++ @typeName(T)),
+    }
+}
+
+test "math.divCeil" {
+    testDivCeil();
+    comptime testDivCeil();
+}
+fn testDivCeil() void {
+    testing.expect((divCeil(i32, 5, 3) catch unreachable) == 2);
+    testing.expect((divCeil(i32, -5, 3) catch unreachable) == -1);
+    testing.expect((divCeil(i32, 5, -3) catch unreachable) == -1);
```

Comment:
prefer `testing.expectEqual` for this sort of test (other code in this file is quite old and pre-dates the introduction of `expectEqual`)

Reply:
Sure thing! I was just afraid to be the only `div` function that used that :worried:

Reply:
I don't think anyone would complain if you felt like updating the whole file ;) (in a separate commit of course)

Reply:
I'll just update my test for now.